sway-lib-std/src/hash.sw

```diff
@@ -304,6 +325,22 @@ pub fn sha256<T>(s: T) -> b256 where T: Hash {
     hasher.sha256()
 }
 
+#![inline(never)]
+pub fn sha256_str<S>(s: S) -> b256 {
+     __check_str_type::<S>();
```

Comment:
The problem with `sha256` and `sha256_str` is that we can´t implement `Hash` for string arrays. That would demand "const generics". Something like that.

```sway
impl<N: const u64> Hash for str[N] {
    ...
}
```

So that is pretty much why we have the "*_str_array" functions and that is why we still need `__check_str_type`; because these functions should only be called with string arrays, and there is no way to express this in the function signature.

I can maybe rename it to `__assert_is_str_array` to make it more clear what it actually does.

Reply:
Renaming it to `__assert_is_str_array` seems like a good idea. Let's see what others thing, though

Reply:
Done.